checkstyle-sonar-plugin/pom.xml

```diff
@@ -111,36 +116,80 @@
       <version>2.0.7</version>
       <scope>test</scope>
     </dependency>
-
+    <dependency>
+      <groupId>org.mockito</groupId>
+      <artifactId>mockito-core</artifactId>
+      <version>2.23.4</version>
+      <scope>test</scope>
+      <exclusions>
+        <exclusion>
+          <artifactId>hamcrest-core</artifactId>
+          <groupId>org.hamcrest</groupId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+    <dependency>
+      <groupId>org.hamcrest</groupId>
+      <artifactId>hamcrest-all</artifactId>
+      <version>1.3</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>junit</groupId>
+      <artifactId>junit</artifactId>
+      <version>4.12</version>
+      <scope>test</scope>
+      <exclusions>
+        <exclusion>
+          <artifactId>hamcrest-core</artifactId>
+          <groupId>org.hamcrest</groupId>
+        </exclusion>
+      </exclusions>
+    </dependency>
     <dependency>
       <groupId>org.sonarsource.sonarqube</groupId>
       <artifactId>sonar-testing-harness</artifactId>
       <scope>test</scope>
       <version>${sonar.version}</version>
     </dependency>
-
     <dependency>
       <groupId>org.easytesting</groupId>
       <artifactId>fest-assert</artifactId>
       <scope>test</scope>
       <version>1.4</version>
     </dependency>
-
     <dependency>
       <groupId>org.slf4j</groupId>
       <artifactId>slf4j-nop</artifactId>
       <!-- 1.5.6(old) version is requied due to sonar 5.6.4 dependency tree -->
       <version>1.5.6</version>
       <scope>test</scope>
     </dependency>
-
     <dependency>
       <groupId>org.apache.ant</groupId>
       <artifactId>ant</artifactId>
       <version>1.9.7</version>
       <scope>test</scope>
     </dependency>
-
+    <dependency>
+      <groupId>com.puppycrawl.tools</groupId>
+      <artifactId>checkstyle</artifactId>
+      <version>${checkstyle.version}</version>
```

Comment:
historically sonar plugin required plugin jar have a conflict with sonar over guava version (sonar used very old version of guava).
That is why we have "checkstyle-all" sub project.

Doe sit become not necessary any more ?

Reply:
I have tried excluding the `checkstyle-all` subproject from all pom files and the entire plugin passes CI and manual tests on SQ 7.5 as well. Hence, I don't really see the purpose of the subproject.
Please check whether we can entirely remove it or additional changes are needed because of this.